### PR TITLE
Add Helm chart release workflow

### PR DESCRIPTION
**Pull request by stefanprodan**
Publish './charts' to gh-pages branch on git tags with 'chart' prefix.

Signed-off-by: stefanprodan <stefan.prodan@gmail.com>

Ref: #1424 

### Release procedure

* create OSM release
* wait for the container image to be pushed to Docker Hub
* set the container image tag in `charts/osm/values.yaml` and `charts/osm/Chart.yaml`
* bump the chart version in `charts/osm/Chart.yaml`
* create a PR with the above changes and merge it to main
* create a git tag from main in the format `chart/vx.x.x` where `vx.x.x` is the chart version
* push the git tag to origin
* GH Actions runs [helm-gh-pages](https://github.com/stefanprodan/helm-gh-pages) that packages the charts, updates `index.yaml` and pushes the changes to `gh-pages` branch
* GH Pages publishes the new version, making the chart available at `https://openservicemesh.github.io/osm`

Note that the above procedure works for any number of charts in the `./charts` dir without any changes to the GH Actions workflow. 

Applicable areas:

- [x] Tests / CI System      

Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?

Yes, I'm the author of the GH Action.

### CODE REVIEW

**Comment by shashankram** [comment]
> Publish './charts' to gh-pages branch on git tags with 'chart' prefix.
> 
> Signed-off-by: stefanprodan [stefan.prodan@gmail.com](mailto:stefan.prodan@gmail.com)
> 
> Ref: #1424
> 
> ### Release procedure
> * create OSM release
> * wait for the container image to be pushed to Docker Hub
> * set the container image tag in `charts/osm/values.yaml` and `charts/osm/Chart.yaml`
> * bump the chart version in `charts/osm/Chart.yaml`
> * create a PR with the above changes and merge it to main
> * create a git tag from main in the format `chart/vx.x.x` where `vx.x.x` is the chart version
> * push the git tag to origin
> * GH Actions runs [helm-gh-pages](https://github.com/stefanprodan/helm-gh-pages) that packages the charts, updates `index.yaml` and pushes the changes to `gh-pages` branch
> * GH Pages publishes the new version, making the chart available at `https://openservicemesh.github.io/osm`
> 
> Note that the above procedure works for any number of charts in the `./charts` dir without any changes to the GH Actions workflow.

Hey @stefanprodan, thanks for this useful change. Would you mind updating `docs/release_guide.md` with these details?

**Comment by stefanprodan** [comment]
@shashankram I've added the chart release procedure to the release guide.

File: docs/release_guide.md [hunk 1]
```diff
@@ -4,19 +4,38 @@ This guide describes the process to create a GitHub Release for this project.
...
+
+In the description section of the new release, add information about feature additions, bug fixes,
+and any other administrative tasks completed on the repository.
+
+## Create a Helm chart release
+
+* set the container image tag in `charts/osm/values.yaml` and `charts/osm/Chart.yaml`
+* bump the chart version in `charts/osm/Chart.yaml`
+* create a PR with the above changes and merge it to main
+* create a Git tag from main in the format `chart/vx.x.x` where `vx.x.x` is the chart version
```
Comment (michelleN) [inline docs/release_guide.md:34]: Right now, the version in our Chart.yaml file is not prefixed with a `v` and I don't usually see the version field in a chart include a `v`. Does this change require use to change the version value in the Chart.yaml file to start with `v`?

Comment (stefanprodan) [reply docs/release_guide.md:34]: No you shouldn't be using `v` in Chart.yaml.

**Comment by stefanprodan** [comment]
> Wondering if we should release the chart as part of the release workflow kicked off by the tag update as well. 

Yes that should work if you also bump the chart version in Chart.yaml. The idea behind having a dedicated tag for charts is for being able to release a chart fix independent of the OSM releases.

File: docs/release_guide.md (hunk 1 above)
Comment (michelleN) [reply docs/release_guide.md:34]: Great. Would you mind removing the `v` from this line? Then, I think we can go ahead and merge. Thank you!

**Pull request merged**